cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -1365,6 +1365,18 @@ TEST(TestBinaryDecimalArithmetic, DispatchBest) {
     }
   }
 
+  // decimal, integer
+  for (std::string name : {"add", "subtract", "multiply", "divide"}) {
+    for (std::string suffix : {"", "_checked"}) {
+      name += suffix;
+
+      CheckDispatchBest(name, {int64(), decimal128(1, 0)},
+                        {decimal128(1, 0), decimal128(1, 0)});
+      CheckDispatchBest(name, {decimal128(1, 0), int64()},
+                        {decimal128(1, 0), decimal128(1, 0)});
```

Comment:
Hmm, so an `int64` is cast to a 1-digit decimal? Unrelated to this PR surely, but I'm a bit surprised :-)

Reply:
I think this is testing that given an integer and a decimal, the integer is cast to a decimal of the same width/precision/scale.

Reply:
But surely the actual cast from int64 to decimal128(1,0) will fail in the precision check?

Reply:
Ah, right. Maybe we want the cast logic to be smarter about that, then. As you said, though, that can be tackled separately.